src/Bitcoin/TransactionBuilder.cpp

```diff
@@ -145,10 +148,16 @@ TransactionPlan TransactionBuilder::plan(const SigningInput& input) {
         } else {
             plan.availableAmount = InputSelector<UTXO>::sum(plan.utxos);
 
+            // There can be less UTXOs after Dust filtering.
+            if (!maxAmount && totalAmount > plan.availableAmount) {
```

Comment:
Shouldn't this also be checked and return an error when `maxAmount == true`?

Reply:
No, because the `total_amount` will be adjusted according to `plan.availableAmount` below at this line:
https://github.com/trustwallet/wallet-core/blob/ff77450cdcb163f1d997cda15dbeefa477191b15/src/Bitcoin/TransactionBuilder.cpp#L189-L190